x/tier/keeper/abci.go

```diff
@@ -0,0 +1,94 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	appparams "github.com/sourcenetwork/sourcehub/app/params"
+	"github.com/sourcenetwork/sourcehub/x/tier/types"
+)
+
+// BeginBlocker claims tier module staking rewards every N blocks.
+// 2% of the claimed rewards is sent to the developer pool (DeveloperPoolFee).
+// 1% is sent to the insurance pool (InsurancePoolFee) if insurance pool balance is below InsurancePoolThreshold,
+// otherwise the InsurancePoolFee (1%) is also sent to the developer pool.
+// Remaining 97% of the rewards is burned.
+func (k *Keeper) BeginBlocker(ctx context.Context) error {
+	params := k.GetParams(ctx)
+
+	// Process rewards every N blocks
+	height := sdk.UnwrapSDKContext(ctx).BlockHeight()
+	if height%params.ProcessRewardsInterval != 0 {
+		return nil
+	}
+
+	tierModuleAddr := authtypes.NewModuleAddress(types.ModuleName)
+	err := k.GetStakingKeeper().IterateDelegations(ctx, tierModuleAddr, func(index int64, delegation stakingtypes.DelegationI) bool {
```

Comment:
suggestion: Although this is obvious if you know how the `IterateDelegations` function or how the staking module stores the delegations works, but lets add a comment here indicating that this iteration loop has a max number of iterations equal to the number of validators. So its bounded :)